interactions.go

```diff
@@ -58,6 +58,35 @@ type ApplicationCommandOptionChoice struct {
 	Value interface{} `json:"value"`
 }
 
+// ApplicationCommandPermissions represents a single user or role permission for a command.
+type ApplicationCommandPermissions struct {
+	ID         string                           `json:"id"`
+	Type       ApplicationCommandPermissionType `json:"type"`
+	Permission bool                             `json:"permission"`
+}
+
+// ApplicationCommandPermissionsList represents a list of ApplicationCommandPermissions, needed for serializing to JSON.
+type ApplicationCommandPermissionsList struct {
+	Permissions []ApplicationCommandPermissions `json:"permissions"`
+}
+
+// GuildApplicationCommandPermissions represents all permissions for a single guild command.
+type GuildApplicationCommandPermissions struct {
+	ID            string                          `json:"id"`
+	ApplicationID string                          `json:"application_id"`
+	GuildID       string                          `json:"guild_id"`
+	Permissions   []ApplicationCommandPermissions `json:"permissions"`
+}
+
+// ApplicationCommandPermissionType indicates whether a permission is user or role based.
+type ApplicationCommandPermissionType uint8
+
+// Application command permission types.
+const (
+	ApplicationCommandPermissionTypeRole = ApplicationCommandPermissionType(iota + 1)
+	ApplicationCommandPermissionTypeUser
```

Comment:
If you could use absolute values rather than `iota` here, that'd be great. In a simple case like this it seems excessive, but `iota` is constantly a source of issues in the library; if new constants are added in the future, and especially if gaps between constants are added (Discord likes to do this), things go wrong quickly. If we have a specific external value that we're depending on, we should use it.